scene/2d/physics_body_2d.cpp

```diff
@@ -1093,9 +1093,26 @@ bool CharacterBody2D::move_and_slide() {
 	floor_normal = Vector2();
 	platform_velocity = Vector2();
 
+	if (motion_mode == MOTION_MODE_GROUNDED) {
+		_move_and_slide_side_logic(motion, current_platform_velocity, prev_floor_normal, prev_platform_rid, prev_platform_layer, was_on_floor);
+	} else {
+		_move_and_slide_top_down_logic(motion);
+	}
```

Comment:
Maybe the methods can be called `_move_and_slide_grounded` and `_move_and_slide_free` to be more explicit about the modes?